### PR TITLE
`pkPaymentErrorForStripeError` no longer returns PKPaymentUnknownErrors

### PR DESCRIPTION
**Pull request by yuki-stripe**
## Summary
`pkPaymentErrorForStripeError` no longer returns PKPaymentUnknownErrors. Instead, it returns the original NSError back, resulting in dismissal of the Apple Pay sheet. 

The Apple Pay sheet doesn't provide useful error messages for PKPaymentUnknownErrors - it's better UX to dismiss the sheet and show the error message to the user yourself.

## Motivation
https://jira.corp.stripe.com/browse/MOBILE-255

## Testing
Forced an error in example VC, saw expected behavior.

### CODE REVIEW

File: Stripe/PublicHeaders/STPAPIClient+ApplePay.h [hunk 1]
```diff
@@ -45,15 +45,16 @@ NS_ASSUME_NONNULL_BEGIN
                             completion:(STPPaymentMethodCompletionBlock)completion;
 
 /**
- Converts Stripe errors into the appropriate Apple Pay error, for use in `PKPaymentAuthorizationResult`. The error is displayed in the Apple Pay sheet, and the user can try again.
+ Converts Stripe errors into the appropriate Apple Pay error, for use in `PKPaymentAuthorizationResult`.
  
- We can convert billing address related errors into a PKPaymentError that helpfully points to the billing address field in the Apple Pay sheet.
- All other errors map to PKPaymentUnknownError, resulting in a generic error message in the Apple Pay sheet.
+ If the error can be fixed by the customer within the Apple Pay sheet, we return an NSError that can be displayed in the Apple Pay sheet.
```
Comment (csabol-stripe) [inline Stripe/PublicHeaders/STPAPIClient+ApplePay.h:50]: based on changelog wouldn't this result in sheet dismissal?

Comment (yuki-stripe) [reply Stripe/PublicHeaders/STPAPIClient+ApplePay.h:50]: Yep. See the ticket for more context, but I think this is the right UX - there's a limited set of errors that can be displayed / are actionable within the Apple Pay sheet.

Comment (csabol-stripe) [reply Stripe/PublicHeaders/STPAPIClient+ApplePay.h:50]: I'm confused about why this says "can be fixed by the customer within the Apple Pay sheet" if the sheet is being dismissed?

Comment (yuki-stripe) [reply Stripe/PublicHeaders/STPAPIClient+ApplePay.h:50]: Updated the README to clarify

**Pull request merged**